src/openforms/submissions/tests/test_read_submission.py

```diff
@@ -116,3 +121,92 @@ def test_retrieve_submission_optional_steps(self):
             ).count(),
             1,
         )
+
+
+class SubmissionReadPaymentInformationTests(SubmissionsMixin, APITestCase):
+    def test_submission_payment_information_no_payment_required(self):
+        submission = SubmissionFactory.create(
+            completed=True,
+            form__generate_minimal_setup=True,
+            form__formstep__optional=True,
+            form__product=None,
+            form__payment_backend="demo",
+        )
+        submission.calculate_price()
+        with self.subTest(part="check data setup"):
```

Comment:
this is pretty neat pattern to separate verifying the test setup and doing the actual tests